### DISCUSSION TITLE
Can I use vim leader in keybindings.json or a when clause in settings.json?

### DISCUSSION THREAD

**Discussion by jeep** [VSCodeVim/Vim#8019] [Q&A]
I would like to bind "RunSelection" to a key using the vim leader key, but it requires some context to know what to run. For example, this works:
Keybindings.json
~~~
      {
        "key": "f19",
        "command": "PowerShell.RunSelection",
        "when": "editorTextFocus && editorLangId == 'powershell'"
      },
      {
        "key": "f19",
        "command": "python.execSelectionInTerminal",
        "when": "editorTextFocus && !findInputFocussed && !jupyter.ownsSelection && !notebookEditorFocused && !replaceInputFocussed && editorLangId == 'python'"
      }
~~~

But this will try to run the selection in both powershell and python (i.e. the when clause doesn't get respected in settings.json)
~~~
    {
      "before": ["<Leader>", "e"],
      "commands": 
      [ 
        { 
          "command": "PowerShell.RunSelection",
          "when": "editorTextFocus && editorLangId == 'powershell'"
        },
        {
          "command": "python.execSelectionInTerminal",
          "when": "editorTextFocus && !findInputFocussed && !jupyter.ownsSelection && !notebookEditorFocused && !replaceInputFocussed && editorLangId == 'python'"
        }
      ]
    },
~~~

Is there a way to use the leader key with a when clause?

**Comment by ttytm** [comment]
hey @jeep,

One way I know of is to map it directly in `keybindings.json`. But you would have to map the key literal. For example, using <kbd>space</kbd> as my leader key it additionally functions as a whichkey initiatior. From there, it can trigger whichkey entries as well as other things I have assigned to the leader key. E.g.,:

```json
	{
		"key": "space",
		"command": "whichkey.show",
		"when": "vim.active && vim.mode != 'Insert' && !inputFocus && !inputBoxFocus && !textInputFocus && !sideBarFocus"
	},
```
